bindings/python/pymongoarrow/api.py

```diff
@@ -352,6 +369,13 @@ def write(collection, tabular):
         )
 
     tabular_gen = _tabular_generator(tabular)
+
+    # Handle Pandas NA objects.
+    codec_options = collection.codec_options
+    if DataFrame is not None:
+        type_registry = TypeRegistry([_PandasNACodec()])
+        codec_options = codec_options.with_options(type_registry=type_registry)
```

Comment:
I'm not sure it's ideal to completely replace the collections type_registry. What if the app already configured a type_registry for other types they want to encode?

Would it be possible to keep the existing registry but add the _PandasNACodec?

Reply:
We could, but we'd have to use private APIs from `TypeRegistry`.

Reply:
https://jira.mongodb.org/browse/ARROW-135